python/sdk/merlin/model.py

```diff
@@ -946,6 +955,43 @@ def deploy(self, environment_name: str = None,
 
         return VersionEndpoint(endpoint, log_url)
 
+    def create_transformer_spec(self, transformer: Transformer, target_env_name: str) -> client.Transformer:
+        resource_request = transformer.resource_request
+        if resource_request is None:
+            env_api = EnvironmentApi(self._api_client)
+            env_list = env_api.environments_get()
+            for env in env_list:
+                if env.name == target_env_name:
+                    resource_request = ResourceRequest(
+                        env.default_resource_request.min_replica,
+                        env.default_resource_request.max_replica,
+                        env.default_resource_request.cpu_request,
+                        env.default_resource_request.memory_request)
+            # This case is when the default resource request is not specified in the environment config
+            if resource_request is None:
+                raise ValueError("resource request must be specified")
+
+        resource_request.validate()
+
+        target_resource_request = client.ResourceRequest(
+            resource_request.min_replica, resource_request.max_replica,
+            resource_request.cpu_request, resource_request.memory_request)
+
+        target_env_vars = []
+        if transformer.env_vars is not None:
+            if not isinstance(transformer.env_vars, dict):
+                raise ValueError(
+                    f"transformer.env_vars should be dictionary, got: {type(transformer.env_vars)}")
+
+            if len(transformer.env_vars) > 0:
+                for name, value in transformer.env_vars.items():
+                    target_env_vars.append(client.EnvVar(name, value))
```

Comment:
Can we handle properly when the `name` and `value` is not string? I think we should add string casting here.

Reply:
Updated